aws/table_aws_cloudtrail_event_data_store.go

```diff
@@ -0,0 +1,184 @@
+			},
+			{
+				Name:        "status",
+				Description: "The status of an event data store.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "created_timestamp",
+				Description: "The timestamp of the event data store's creation.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "multi_region_enabled",
+				Description: "Indicates whether the event data store includes events from all regions, or only from the region in which it was created.",
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "organization_enabled",
+				Description: "Indicates that an event data store is collecting logged events for an organization.",
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "retention_period",
+				Description: "The retention period, in days.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "termination_protection_enabled",
+				Description: "Indicates whether the event data store is protected from termination.",
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "updated_timestamp",
+				Description: "The timestamp showing when an event data store was updated, if applicable. UpdatedTimestamp is always either the same or newer than the time shown in CreatedTimestamp.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+			{
+				Name:        "advanced_event_selectors",
+				Description: "The advanced event selectors that were used to select events for the data store.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCloudTrailEventDataStore,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("EventDataStoreArn").Transform(transform.EnsureStringArray),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listCloudTrailEventDataStores(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Get client
+	svc, err := CloudTrailClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Info("aws_cloudtrail_event_data_store.listCloudTrailEventDataStores", "client_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
```

Comment:
I think in the later Steampipe CLI versions if someone runs a query with `limit 0`, the hydrate functions aren't entered, so we may not need this check, so we could simplify to:
```
		if limit < maxLimit {
			maxLimit = limit
		}
```

Can you please test and confirm this?